lib/maintenance_tasks.rb

```diff
@@ -32,6 +32,18 @@ module MaintenanceTasks
   #   the ticker during Task iterations.
   mattr_accessor :ticker_delay, default: 1.second
 
+  # Retrieves the callback to be performed when an error occurs in the task.
+  mattr_reader :error_handler,
+    default: ->(_error, _task_context, _errored_element) {}
+
   # Defines a callback to be performed when an error occurs in the task.
-  mattr_accessor :error_handler, default: ->(_error) {}
+  def self.error_handler=(error_handler)
+    unless error_handler.arity.equal?(3)
+      ActiveSupport::Deprecation.warn(
+        'MaintenanceTasks.error_handler be an object that responds to .call '\
+         'and takes three arguments: error, task_context, and errored_element.'
+      )
+    end
+    @@error_handler = error_handler # rubocop:disable Style/ClassVars
```

Comment:
`MaintenanceTasks` is not going to be inherited, so we could be using standard instance variables for the Module instance. That would require changing the reader but would avoid the rubocop disable here.